sw/device/lib/dif/dif_hmac.h

```diff
@@ -0,0 +1,138 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 1.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_LIB_DIF_HMAC_DEVICE_H_
+#define SW_DEVICE_LIB_DIF_HMAC_DEVICE_H_
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+
+/**
+ * Supported HMAC operations.
+ */
+typedef enum dif_hmac_ops {
+  DIF_HMAC_OP_HMAC = 0,
+  DIF_HMAC_OP_SHA256 = 1
+} dif_hmac_ops_t;
+
+/**
+ * Configuration for initializing the HMAC device.
+ */
+typedef struct dif_hmac_config {
+  /** The base address for registers in the HMAC IP*/
+  mmio_region_t base_addr;
+  /** The mode of operation */
+  dif_hmac_ops_t mode;
+  /** Message endianness. true -> big endian, false -> little endian. */
+  bool endian_swap;
+  /** Digest endianness. true -> big endian, false -> little endian. */
+  bool digest_swap;
+  /** The HMAC key. Only used in |DIF_HMAC_OP_HMAC| mode. */
+  uint32_t key[8];
+} dif_hmac_config_t;
+
+/**
+ * A typed representation of the HMAC digest.
+ */
+typedef struct dif_hmac_digest { uint32_t digest[8]; } dif_hmac_digest_t;
```

Comment:
Could you run `git clang-format` (I think only considers `git add` added files) or `clang-format` on all changed files.

Reply:
This formatting was actually produced by clang-format on the Azure pipeline. My local clang-format doesn't produce this, but I was taking the Azure pipeline checks as the source of truth.

```
diff --git a/sw/device/lib/dif/dif_hmac.h b/sw/device/lib/dif/dif_hmac.h
index 25175a8e..7288c5f1 100644
--- a/sw/device/lib/dif/dif_hmac.h
+++ b/sw/device/lib/dif/dif_hmac.h
@@ -37,9 +37,7 @@ typedef struct dif_hmac_config {
 /**
  * A typed representation of the HMAC digest.
  */
-typedef struct dif_hmac_digest {
-  uint32_t digest[8];
-} dif_hmac_digest_t;
+typedef struct dif_hmac_digest { uint32_t digest[8]; } dif_hmac_digest_t;
 
 /**
  * State for a particular HMAC device.
@@ -47,9 +45,7 @@ typedef struct dif_hmac_digest {
  * its member variables should be considered private, and are only provided so
  * that callers can allocate it.
  */
-typedef struct dif_hmac {
-  mmio_region_t base_addr;
-} dif_hmac_t;
+typedef struct dif_hmac { mmio_region_t base_addr; } dif_hmac_t;
 
 /**
  * Initializes the SPI device described by |config|, writing internal state to
```

Reply:
I think this is a problem where the clang-format on our workstations might have slightly different defaults. I suspect Azure is just using an older clang-format. *shrug*

Reply:
Seems that way. Azure and my `clang-format` disagree in this specific case, and I've been manually fixing this to appease Azure.